include/aktualizr-lite/aklite_client_ext.h

```diff
@@ -0,0 +1,30 @@
+#ifndef AKTUALIZR_LITE_CLIENT_EXT_H_
+#define AKTUALIZR_LITE_CLIENT_EXT_H_
+
+#include <cstdlib>
+#include <string>
+
+#include "json/json.h"
+
+#include "aktualizr-lite/tuf/tuf.h"
+
+#include "aktualizr-lite/api.h"
+
+/**
+ * @brief contains additional methods that consolidate functionality
+ * making it reusable between the main aklite daemon and other tools.
+ *
+ * Those methods may eventually be part of the supported API but, for now,
+ * they are likely to have they signature and behavior changed.
+ */
+class AkliteClientExt : public AkliteClient {
+  // Inherit constructors
+  using AkliteClient::AkliteClient;
+
+ public:
+  CheckInResult GetTargetToInstall(const LocalUpdateSource* local_update_source, int version,
+                                   const std::string& target_name, const TufTarget& current, bool allow_bad_target,
+                                   bool force_apps_sync);
```

Comment:
I would call it something like `allow_current_if_not_sync`.
I would probably introduce a bitwise `enum` to define type of targets to include into the target search and return.

Reply:
The method always returns the current target if the apps are not in sync. That's the behavior of the original daemon, it checks if apps are in sync on every iteration, doing an install if they are not.
`force_apps_sync=true` means that we return the current one even if the apps are in sync. That's the behavior for offline updates.